examples/exporter/logexporter.go

```diff
@@ -0,0 +1,190 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package exporter contains a log exporter that supports exporting
+// OpenCensus metrics to a logging framework.
+package exporter // import "go.opencensus.io/examples/exporter"
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/metric/metricexport"
+	"go.opencensus.io/trace"
+)
+
+// LogExporter exports stats to log file
+type LogExporter struct {
+	reader         *metricexport.Reader
+	ir             *metricexport.IntervalReader
+	initReaderOnce sync.Once
+	o              Options
+	tFile          *os.File
+	mFile          *os.File
+	tLogger        *log.Logger
+	mLogger        *log.Logger
+}
+
+// Options provides options for LogExporter
+type Options struct {
+	// ReportingInterval is a time interval between two successive metrics
+	// export.
+	ReportingInterval time.Duration
+
+	// MetricsLogFile is path where exported metrics are logged.
+	// If it is nil then the metrics are logged on console
+	MetricsLogFile string
+
+	//TracesLogFile is path where exported span data are logged.
+	// If it is nil then the span data are logged on console
+	TracesLogFile string
+}
+
+func getLogger(filepath string) (*log.Logger, *os.File) {
+	if filepath == "" {
+		return log.New(os.Stdout, "", 0), nil
+	}
+	f, err := os.OpenFile(filepath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		log.Fatalf("error opening file: %v", err)
+	}
+	return log.New(f, "", 0), f
+}
+
+// NewLogExporter creates new log exporter.
+func NewLogExporter(options Options) (*LogExporter, error) {
+	e := &LogExporter{reader: metricexport.NewReader(),
+		o: options}
+	e.tLogger, e.tFile = getLogger(options.TracesLogFile)
+	e.mLogger, e.mFile = getLogger(options.MetricsLogFile)
+	return e, nil
+}
+
+func printMetricDescriptor(metric *metricdata.Metric) string {
+	d := metric.Descriptor
+	return fmt.Sprintf("name: %s, type: %s, unit: %s ",
+		d.Name, d.Type, d.Unit)
+}
+
+func printLabels(metric *metricdata.Metric, values []metricdata.LabelValue) string {
+	d := metric.Descriptor
+	kv := []string{}
+	for i, k := range d.LabelKeys {
+		kv = append(kv, fmt.Sprintf("%s=%v", k, values[i]))
+	}
+	return fmt.Sprintf("%v", kv)
+}
+
+func printPoint(point metricdata.Point) string {
+	switch v := point.Value.(type) {
+	case *metricdata.Distribution:
+		dv := v
+		return fmt.Sprintf("count=%v sum=%v sum_sq_dev=%v, buckets=%v", dv.Count,
+			dv.Sum, dv.SumOfSquaredDeviation, dv.Buckets)
+	default:
+		return fmt.Sprintf("value=%v", point.Value)
+	}
+}
+
+// Start starts the metric exporter.
```

Comment:
Nit: metric and trace

Reply:
it doesn't start trace exporter. For trace it sill requires to register but I don't see why I cannot call registerExporter here. I'll make the change  and update the comment.